src/System.Runtime.Extensions/src/Resources/Strings.resx

```diff
@@ -166,6 +166,9 @@
   <data name="Arg_AppDomainUnloadedException" xml:space="preserve">
     <value>Attempted to access an unloaded AppDomain.</value>
   </data>
+  <data name="Arg_SwitchExpressionException" xml:space="preserve">
+    <value>Non-exhaustive switch expression failed to match its input.</value>
```

Comment:
Can we include UnmatchedValue.ToString() in the error message if it's non-null?

Reply:
Have to be careful putting `ToString()` of an arbitrary object into a message. It can be huge. I suggest using ellipses, like this example: https://github.com/dotnet/corefx/blob/2d9dd533e16889ea4d279162df4039982bfbbca2/src/System.Text.Json/src/System/Text/Json/ThrowHelper.cs#L350
Pick a length, I see 32 used elsewhere.

Reply:
> Have to be careful putting ToString() of an arbitrary object into a message. It can be huge

How is this any different from putting arbitrary paths into file exceptions (paths can be thousands of characters), or ArgumentOutOfRangeException including _actualValue.ToString() in the message:
https://github.com/dotnet/coreclr/blob/b2e3a19be00af54c3bc4c76d94044aca9c32a45d/src/System.Private.CoreLib/shared/System/ArgumentOutOfRangeException.cs#L82
?

I'd err on the side of including the whole thing.  The point of this exception is to highlight a bug in the code that the compiler wasn't able to flag, and the purpose of the UnmatchedValue is to give you as much information as possible to fix that bug.  Trimming it to some short length is likely to limit its benefit, especially if ToString wasn't overridden and thus results in rendering the full type name, which has a decent chance of being longer than 32 characters.

Reply:
Fine. Might be good to put the value at the end of the message not the middle though.

Reply:
Sounds good. thanks.